src/server/game/Miscellaneous/Formulas.h

```diff
@@ -176,7 +176,10 @@ namespace Trinity
             {
                 float xpMod = 1.0f;
 
-                gain = BaseGain(player->getLevel(), u->getLevel());
+                if (creature && creature->HasScalableLevels())
```

Comment:
Instead of doing an if for all things, i would change all getLevels to GetlevelForTarget and then let the virtual override handle the rest.. put the scaling if in the GetLevelForTarget in the creature override and let it handle there.